File: pref/theme/theme_android.java
package org.gioui.x.pref.theme;

import android.content.res.Resources;
import android.content.res.Configuration;
import android.provider.Settings.Global;
import android.os.Build;

public class theme_android {
	public static int isDark() {
        Resources res = Resources.getSystem();
        if (res == null) {
            return -1;
        }

        Configuration config = res.getConfiguration();
        if (config == null) {
            return -1;
        }

        if ((config.uiMode & Configuration.UI_MODE_NIGHT_MASK) == Configuration.UI_MODE_NIGHT_YES) {
           return 1;
        }

        return 0;
	}

	public static int isReducedMotion() {
	    if (Global.TRANSITION_ANIMATION_SCALE == "0") {
	        return 1;
	    }
	    return 0;
	}
}

File: pref/theme/theme_android.go
package theme

//go:generate javac -source 8 -target 8 -bootclasspath $ANDROID_HOME/platforms/android-29/android.jar -d $TEMP/pref_theme/classes theme_android.java
//go:generate jar cf theme_android.jar -C $TEMP/pref_theme/classes .

import (
	"gioui.org/x/pref/internal/xjni"
)

var _Lib = "org/gioui/x/pref/theme/theme_android"

func isDark() (bool, error) {
	i, err := xjni.DoInt(_Lib, "isDark", "()I")
	if err != nil || i < 0 {
		return false, ErrNotAvailableAPI
	}
	return i >= 1, nil
}

func isReducedMotion() (bool, error) {
	i, err := xjni.DoInt(_Lib, "isReducedMotion", "()I")
	if err != nil || i < 0 {
		return false, ErrNotAvailableAPI
	}
	return i >= 1, nil
}
